### DISCUSSION TITLE
More Control and Optimised DOM Output

### DISCUSSION THREAD

**Discussion by Daveden2** [elementor/elementor#22259] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I'm frustrated when I want to create a complex accessible list or table and have to rely on third-party plugins or write out the entire HTML myself. 
When we want to add accessible ARIA labels like "aria-labelledby" to a `<ul>`, it gets added to the wrapper div instead of the main `<ul>` element.

### Describe the solution you'd like

It would be nice if we had the following:

1. A toggle to strip out all unnecessary div wrappers similar to the Unwrap Feature offered by Dynamicooo's. Bricks builder has something similar with their code element where we have a toggle to "execute code without wrapper".
2. An option to specify where custom attributes are applied, as it currently applies to the wrapper `<div>` instead of the desired `<ul>` etc. So we currently cannot do apply `aria-label` or `aria-labelledby` on the Icon List widget.
3. Custom HTML tags for Containers element where we can define custom HTML tags e.g. `<picture>`, `<ul>`, `<li>`, `<address>` etc to build customisable layouts.
4. Also a blank/basic text element without wrapper divs that can be set to any HTML tag - `<li>`, `<th>`, `<tr>`, `<a>`, `<button>`, `<source>` etc.

### Describe alternatives you've considered

I can do the unwrap of unnecessary divs using Dynamicooo's Unwrap feature but unfortunately, I no longer have the ability to add custom attributes.
I can go with another builder like Bricks Builder, which allows custom tags and doesn't add unnecessary wrapper divs to elements

**Comment by Bovelett** [comment]
Big plus one, especially on the attribute request. Would be a great step towards being accessibility-ready.

**Comment by SpinDreams** [comment]
+1 this is defo needed. Lean and mean.

**Comment by ayoubkhan558** [comment]
+1

**Comment by ayoubkhan558** [comment]
Elementor 3.20 beta was released with a reduced DOM version.

[🔥 Elementor v3.20 Beta Release](https://github.com/orgs/elementor/discussions/25200)  #25200

**Reply by Daveden2** [reply]
@ayoubkhan558 It's a start. It removed a few html attributes from Containers but it hasn't removed the extra **wrapper divs** from Widgets
